File: CampingNeretva/CampingNeretva.Model/ImageModels/ImageModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CampingNeretva.Model.ImageModels
{
    public class ImageModel
    {
        public int ImageId { get; set; }
        public string Path { get; set; }
        public DateTime DateCreated { get; set; }
        public string ContentType { get; set; }
    }
}

File: CampingNeretva/UI/campingneretva_desktop/lib/models/image_model.dart
class ImageModel {
  final int imageId;
  final String path;
  final String contentType;
  final DateTime dateCreated;

  ImageModel({
    required this.imageId,
    required this.path,
    required this.contentType,
    required this.dateCreated,
  });

  factory ImageModel.fromJson(Map<String, dynamic> json) {
    return ImageModel(
      imageId: json['imageId'],
      path: json['path'],
      contentType: json['contentType'],
      dateCreated: DateTime.parse(json['dateCreated']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'imageId': imageId,
      'path': path,
      'contentType': contentType,
      'dateCreated': dateCreated.toIso8601String(),
    };
  }
}
